src/client/customers.rs

```diff
@@ -235,6 +235,203 @@ pub struct Address {
+    pub per_unit_cost_basis: Option<String>,
+}
+
+/// The type of ledger entry
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(tag = "entry_type")]
+pub enum LedgerEntry {
+    /// Incrementing a credit balance
+    #[serde(rename = "increment")]
+    Increment(IncrementLedgerEntry),
+    /// Voiding of an existing ledger entry
+    #[serde(rename = "void")]
+    Void(VoidLedgerEntry),
+    /// Voiding of an existing ledger entry has been initiated
+    #[serde(rename = "void_initiated")]
+    VoidInitiated(VoidInitiatedLedgerEntry),
+    // TODO: additional ledger entry types
+}
+
+/// The state of a ledger entry
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize_enum_str)]
+pub enum EntryStatus {
+    /// The entry has been committed to the ledger
+    #[serde(rename = "committed")]
+    Committed,
+    /// The entry hasn't yet been committed to the ledger
+    #[serde(rename = "pending")]
+    Pending,
+}
+
+/// A collection of identifiers for a customer
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct CustomerIdentifier {
+    /// The Orb-assigned unique identifier for the customer.
+    pub id: String,
+    /// An optional user-defined ID for this customer resource, used throughout
+    /// the system as an alias for this customer.
+    pub external_customer_id: Option<String>,
+}
+
+/// Credit block data associated with entries in a ledger.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct LedgerEntryCreditBlock {
+    //// The Orb-assigned unique identifier for the credit block.
+    pub id: String,
+    /// The date on which the block's balance will expire.
+    #[serde(with = "time::serde::rfc3339::option")]
+    pub expiry_date: Option<OffsetDateTime>,
+    /// The price per credit.
+    pub per_unit_cost_basis: Option<String>,
+}
+
+/// Core ledger entry fields.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct BaseLedgerEntry {
+    /// The Orb-assigned unique identifier for the ledger entry.
+    pub id: String,
+    /// An incrementing identifier ordering the ledger entry relative to others.
+    pub ledger_sequence_number: u64,
+    /// The state of the ledger entry.
+    pub entry_status: EntryStatus,
+    /// The customer identifiers associated with the ledger entry.
+    pub customer: CustomerIdentifier,
+    /// The customer's credit balance before application of the ledger operation.
+    pub starting_balance: serde_json::Number,
+    /// The customer's credit balance after application of the ledger operation.
+    pub ending_balance: serde_json::Number,
+    /// The amount granted to the ledger.
+    pub amount: serde_json::Number,
+    /// The date the ledger entry was created.
+    #[serde(with = "time::serde::rfc3339")]
+    pub created_at: OffsetDateTime,
+    /// An optional description to associate with the entry.
+    pub description: Option<String>,
```

Comment:
similar, do you want to skip if none?

Reply:
Same comment re: this only being used when deserializing an API response.